src/qibolab/pulses/sequence.py

```diff
@@ -21,16 +21,20 @@ def __init__(self, seq_dict: Optional[dict[str, list[PulseLike]]] = None):
     def probe_pulses(self):
         """Return list of the readout pulses in this sequence."""
         pulses = []
-        for seq in self.values():
+        for name, seq in self.items():
+            if "probe" not in name:
```

Comment:
You are probably aware, but I think this is a bit error-prone because channel names are defined by the guy writing the platform (=lab user) and we cannot assume they will always use names that make sense, for example, what if they use the usual L-627, etc.. 

This could be an issue here, because even if we introduce a new `ProbeChannel(IqChannel)` type for probe channels (which is reasonable given that we have an `AcquireChannel`), the `PulseSequence` will not have information about this since it only has its name as `str`. On the other hand, we could solve the issue by dropping the `probe_pulses` property altogether. Is it needed anywhere?

The general question is does the `PulseSequence` really needs to know anything about pulse types? I think no, because the entities that are playing with it (`Platform` and instruments) already have information about channels and configs and can use it to devise the pulse type if needed. For example in QM this is how I identify is a `Pulse` is `PulseType.READOUT`: https://github.com/qiboteam/qibolab/blob/142e09888b0a67144b249c559d6750398d61e0a2/src/qibolab/instruments/qm/controller.py#L274-L277

Reply:
> You are probably aware, but I think this is a bit error-prone because channel names are defined by the guy writing the platform (=lab user) and we cannot assume they will always use names that make sense, for example, what if they use the usual L-627, etc..

Yes, I know, and I don't like this solution. But it's just a temporary one.
My idea is to reintroduce the useful information about the pulse type somehow. But not much in the pulse, but rather in the channel.

In #965 I already defined a `ChannelId`, which for the time being is still a `str`, but I wanted to upgrade it to wrap the information about the channel type.
This would duplicate some information, but the minimal amount, since a `PulseSequence` may exist outside a platform.

The other option (and maybe the best one) is to give up on attributing this feature to the sequence, and instead move it to the platform (which has also access to the channels, and can verify their types).

> This could be an issue here, because even if we introduce a new `ProbeChannel(IqChannel)` type for probe channels (which is reasonable given that we have an `AcquireChannel`), the `PulseSequence` will not have information about this since it only has its name as `str`. On the other hand, we could solve the issue by dropping the `probe_pulses` property altogether. Is it needed anywhere?
> 
> The general question is does the `PulseSequence` really needs to know anything about pulse types? I think no, because the entities that are playing with it (`Platform` and instruments) already have information about channels and configs and can use it to devise the pulse type if needed. For example in QM this is how I identify is a `Pulse` is `PulseType.READOUT`:
> 
> [](https://github.com/qiboteam/qibolab/blob/142e09888b0a67144b249c559d6750398d61e0a2/src/qibolab/instruments/qm/controller.py#L274-L277)

Whoops, I wrote the part above before checking the rest. And yes, I agree with it. It's just a matter of propagating the information in the driver (as I described above, there is the problem that most of the drivers are taking pulses from anywhere, e.g. sweepers, so the relation with the channel might be lost)

Reply:
As discussed, I think we will eventually just drop this property.

Reply:
Already dropped in #965. Still handling the aftermath in there